util/network-devp2p/src/discovery.rs

```diff
@@ -574,7 +583,7 @@ impl<'a> Discovery<'a> {
 		Ok(None)
 	}
 
-	fn is_a_valid_known_node(&self, node: &NodeEntry) -> bool {
+	fn is_a_valid_known_node(&mut self, node: &NodeEntry) -> bool {
 		let id_hash = keccak(node.id);
 		let dist = match Discovery::distance(&self.id_hash, &id_hash) {
 			Some(dist) => dist,
```

Comment:
Doesn't returning `false` in case of ourselves cause a ping to be sent anyway?

Reply:
Good catch! Will make it an enum